certbot-dns-google/certbot_dns_google/dns_google_test.py

```diff
@@ -50,6 +52,16 @@ def test_cleanup(self):
         expected = [mock.call.del_txt_record(DOMAIN, '_acme-challenge.'+DOMAIN, mock.ANY, mock.ANY)]
         self.assertEqual(expected, self.mock_client.mock_calls)
 
+    @mock.patch('httplib2.Http.request', side_effect=ServerNotFoundError)
+    def test_without_auth(self, unused_mock):
+        from certbot_dns_google.dns_google import Authenticator
+
+        super(AuthenticatorTest, self).setUp()
+        self.config = mock.MagicMock(google_credentials=None)
```

Comment:
What I personally had in mind here was just changing this to:
```
    @mock.patch('httplib2.Http.request', side_effect=ServerNotFoundError)
    def test_without_auth(self, unused_mock):
        self.config.google_credentials = None
        self.assertRaises(PluginError, self.auth.perform, [self.achall])
```
This works because `self.config` refers to the same object in the authenticator. You can also do `self.auth.config.google_credentials = None` which may make this cleaner/clearer.

I just think we can reuse the `self.auth` object already created for us by the `setUp` function.